### PR TITLE
[erc721-with-metadata-by-ownerof] Implemented erc721-with-metadata-simple strategy

### PR DESCRIPTION
**Pull request by victor-kyriazakos**
Changes proposed in this pull request:

Added "erc721-with-metadata-simple" strategy
Similar to "erc721-tokenid-range-weights-simple", it does not require the erc721Enumerable extension
Does not use "tokenOfOwnerByIndex" contract method

### CODE REVIEW

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi @victor-kyriazakos What's the difference between this strategy and https://snapshot.org/#/strategy/erc721-with-metadata strategy? 

Also revert all the changes from other files than your new strategies including package-lock and yarn.lock

**Comment by victor-kyriazakos** [comment]
Hi @ChaituVR I reverted the changes you asked for.

As described in the original description, the difference with "erc721-with-metadata" is that it doesn't require the contract to implement the ERC721Enumerable interface by OpenZeppeling, similar to the difference between "erc721-with-tokenid-range-weights" and "erc721-with-tokenid-range-weights-simple". It does not use the "tokenOfOwnerByIndex" contract method

**Review by ChaituVR — CHANGES_REQUESTED** [review]
```/home/runner/work/snapshot-strategies/snapshot-strategies/src/strategies/erc721-with-metadata-simple/index.ts
Error:   17:9  error  'ids' is never reassigned. Use 'const' instead  prefer-const```

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi @victor-kyriazakos The name doesn't match the strategy, it is def. not as `simple` as the existing strategy 😅 Can you think of a different name? maybe like `erc721-with-metadata-by-ownerof`?

File: src/strategies/erc721-with-metadata-simple/index.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import { multicall } from '../../utils';
+import snapshots from '@snapshot-labs/snapshot.js';
+
+export const author = 'victor-kyriazakos';
+export const version = '0.1.0';
+
+const abi = [
+  'function balanceOf(address account) external view returns (uint256)',
```
[the remark this answers is not in the document]
Comment (victor-kyriazakos) [reply src/strategies/erc721-with-metadata-simple/index.ts]: Correct, not used - removed

File: src/strategies/erc721-with-metadata-simple/index.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  //const { tokenIdWeightRanges, defaultWeight } = options;
+  const maximumNumberOfBatches = 4;
+  const batchSize = 8000;
+  const maximumAllowedRange = maximumNumberOfBatches * batchSize;
```
[the remark this answers is not in the document]
Comment (victor-kyriazakos) [reply src/strategies/erc721-with-metadata-simple/index.ts]: Defined the maxAllowedRange to 8000 * 4 = 32000 directly. Let me know if you meant something different here

**Comment by victor-kyriazakos** [comment]
Hi @ChaituVR I changed the strategy name to your suggestion - `erc721-with-metadata-by-ownerof` . I think the `-simple` reflected that it works for "simple erc721 contracts" that do not implement the full IEnumerable interface, but your suggested name works just fine :D

File: src/strategies/erc721-with-metadata-by-ownerof/index.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    // Define contract calls
+    const contractCalls = ids.map((id: number) => [
+      options.address,
+      'ownerOf',
+      [id]
+    ]) as [string, string, [number]][];
+
+    // batch-call contract data
+    const customRangeResponse = await getDataFromBlockChain(contractCalls);
+    const customRangeResponseWeighted = await multiplyOccurrencesByWeights(
```
Comment (ChaituVR) [inline src/strategies/erc721-with-metadata-by-ownerof/index.ts:95]:
```suggestion
    const customRangeResponseWeighted = multiplyOccurrencesByWeights(
```

File: src/strategies/erc721-with-metadata-by-ownerof/index.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  ) => multicall(network, provider, abi, contractCalls, { blockTag });
+
+  const filterUnusedAddresses = (addressesToFilter: string[]): string[] =>
+    addressesToFilter.filter((address: string) =>
+      addresses
+        .map((address: string) => address.toLowerCase())
+        .includes(address.toLowerCase())
+    );
+
+  const multiplyOccurrencesByWeights = async (
```
Comment (ChaituVR) [inline src/strategies/erc721-with-metadata-by-ownerof/index.ts:66]:
```suggestion
  const multiplyOccurrencesByWeights = (
```

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**